### DISCUSSION TITLE
Lottery(Interactions.s.sol)

### DISCUSSION THREAD

**Discussion by yasharyan0904** [Cyfrin/foundry-full-course-cu#1885] [Q&A] [answered]
How does the the following code works? I mean use of abi.encode(subId)!
```
vm.startBroadcast(deployerKey);
            LinkToken(link).transferAndCall(
                vrfCoordinatorV2,
                FUND_AMOUNT,
                abi.encode(subId)
            );
            vm.stopBroadcast();
```            
The code for LinkToken
```
function transferAndCall(
        address _to,
        uint256 _value,
        bytes memory _data
    ) public virtual returns (bool success) {
        super.transfer(_to, _value);
        // emit Transfer(msg.sender, _to, _value, _data);
        emit Transfer(msg.sender, _to, _value, _data);
        if (isContract(_to)) {
            contractFallback(_to, _value, _data);
        }
        return true;
    }
```

**Comment by EngrPips** [comment, marked as answer]
Hello @yasharyan0904, The `transferAndCall` function is expecting an argument which is of type `bytes` as specified in the function `bytes memory _data` so in order to convert `subId` which is naturally a type of uint256 we had to do the `abi.encode(subId)` and like that we have a data which is of `bytes` type that we can pass into the the `transferAndCall` function. If anything is confusing please ask questions.

**Reply by yasharyan0904** [reply]
Ok thanks. 
Just a question out of curiosity, why we have to convert it into bytes? Even in RaffleTest.t.sol later abi is used!! Why bytes is preferred??

**Reply by EngrPips** [reply]
We have to convert the `subId` into bytes because the function we want to call expects an argument which is of type `bytes`  and the reason why bytes is preferred I can't really say. I think need to do some research on that.

**Reply by yasharyan0904** [reply]
THANKS:)

**Reply by EngrPips** [reply]
You are welcome, Friend.
